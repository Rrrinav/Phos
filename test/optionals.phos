model Item { let name: string; }

fn get_item(id: i64) -> Item? {
    if (id == 1) {
        return Item{ .name = "Phos" };
    }
    return nil;
}

fn process_item(item: Item) {
    print("Processing: " + item.name);
}

let item1 := get_item(1);
let item2 := get_item(2);

if (item1 != nil) {
    print("Item 1 is not nil.");
    process_item(item1 as Item); // Safe cast
}

if (item2 == nil) {
    print("Item 2 is nil.");
    // process_item(item2 as Item); // Would be a type error
}
