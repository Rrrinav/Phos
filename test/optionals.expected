Item 1 is not nil.
Processing: Phos
Item 2 is nil.
